File: src/test/java/de/weimarnetz/registrator/services/NetworkVerificationServiceTest.java
package de.weimarnetz.registrator.services;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.when;

import java.util.Map;

import org.assertj.core.util.Maps;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;

import de.weimarnetz.MockitoTest;
import de.weimarnetz.registrator.configuration.NetworkInformation;
import de.weimarnetz.registrator.configuration.NetworksConfiguration;

public class NetworkVerificationServiceTest extends MockitoTest {

    @InjectMocks
    private NetworkVerificationService networkVerificationService;
    @Mock
    private NetworksConfiguration networksConfiguration;

    @BeforeEach
    public void setup() {
        Map<String, NetworkInformation> networkInformationMap = Maps.newHashMap("testnet", null);
        when(networksConfiguration.getMap()).thenReturn(networkInformationMap);
    }

    @Test
    public void invalidNetworkTest() {
        // when
        boolean networkValid = networkVerificationService.isNetworkValid("not_our_netz");

        // then
        assertThat(networkValid).isFalse();
    }

    @Test
    public void validNetworkTest() {
        // when
        boolean networkValid = networkVerificationService.isNetworkValid("testnet");

        // then
        assertThat(networkValid).isTrue();
    }
}

File: src/test/java/de/weimarnetz/registrator/services/NetworkVerificationServiceTest.kt
package de.weimarnetz.registrator.services

import de.weimarnetz.registrator.configuration.NetworkInformation
import de.weimarnetz.registrator.configuration.NetworksConfiguration
import io.mockk.every
import io.mockk.mockk
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class NetworkVerificationServiceTest {
    private val networksConfiguration: NetworksConfiguration = mockk()

    private val networkVerificationService = NetworkVerificationService(networksConfiguration)

    @BeforeEach
    fun setup() {
        val networkInformationMap = mapOf("testnet" to NetworkInformation())
        every { networksConfiguration.map } returns networkInformationMap
    }

    @Test
    fun invalidNetworkTest() {
        // when
        val networkValid = networkVerificationService.isNetworkValid("not_our_netz")

        // then
        assertThat(networkValid).isFalse
    }

    @Test
    fun validNetworkTest() {
        // when
        val networkValid = networkVerificationService.isNetworkValid("testnet")

        // then
        assertThat(networkValid).isTrue
    }
}